modules/ocl/cl_kernel.cpp

```diff
@@ -111,19 +119,75 @@ CLKernel::build_kernel (const XCamKernelInfo& info, const char* options)
 
     {
         SmartLock locker (_kernel_map_mutex);
+
+        char* cache_path = std::getenv ("XCAM_CL_KERNEL_CACHE_PATH");
+        if (cache_path != NULL) {
+            if (access (cache_path, F_OK) == -1) {
+                mkdir (cache_path, S_IRWXU | S_IRWXG | S_IROTH | S_IXOTH);
+            }
+            memset (_kernel_cache_path, 0, XCAM_MAX_STR_SIZE);
+            snprintf (
+                _kernel_cache_path, XCAM_MAX_STR_SIZE - 1,
+                "%s/%s",
+                cache_path, key_str);
+        } else {
+            if (access (_kernel_cache_path, F_OK) == -1) {
+                mkdir (_kernel_cache_path, S_IRWXU | S_IRWXG | S_IROTH | S_IXOTH);
+            }
+            strncat (_kernel_cache_path, key_str, XCAM_MAX_STR_SIZE - 1);
```

Comment:
kernel file name, how about use
kernel_cache_path + kernel_name + key_id.
human readable file name can be easy for debug.